taker-frontend/src/components/Tour.tsx

```diff
@@ -0,0 +1,182 @@
+        },
+        {
+            element: "#longLeverage",
+            // TODO: Add link to learn about leverage trading
+            intro: "The leverage influences the margin as well. At the moment the leverage is fixed at x2.",
+            position: "right",
+        },
+        {
+            element: "#longRequiredMargin",
+            intro:
+                "This is the amount of BTC that is necessary to open the position. This amount will be locked on chain.",
+            position: "right",
+        },
+        {
+            element: "#longPerpetualCost",
+            intro:
+                "To allow you to close any point in time in the future you pay a small fee per the hour. Initially you will pay fees for 24h, with every hour that the CFD remains open you pay for one additional hour.",
+            position: "right",
+        },
+        {
+            element: "#longButton",
+            intro: (
+                <>
+                    <Text>
+                        Clicking this button will open a confirmation dialog similar to this:
+                        <br />
+                        <img src={confirmationDialog} width={"400px"} alt={"Confirm it already!"} />
+                        <br />
+                        Once you confirm the CFD will be opened with the maker, resulting in your and the maker's margin
+                        being locked up on chain.
+                        <br />
+                        <br />
+                        You can close an open position at any point in time.
+                    </Text>
+                </>
+            ),
+            position: "right",
+        },
+        {
+            title: "Time for funding the wallet!",
+            element: "#walletSwitchButton",
+            intro: "Time to add some funds to the wallet... Let me take you there...",
+            position: "right",
+        },
+        {
+            title: "Happy trading!",
+            intro: (
+                <>
+                    <Text>
+                        On Umbrel the wallet is derived from your Umbrel seed. For details check the&nbsp;<FaqLink />
+                        <br />
+                        <br />
+                        Send BTC to the address below. A new address will be derived after usage. Balance is picked up
+                        once the tx is seen in mempool. You can then open a position.
+                    </Text>
+                </>
+            ),
+            position: "right",
+        },
+    ];
+
+    const [tourEnabled, setTourEnabled] = useState(true);
+
+    const onExit = () => {
+        setTourEnabled(false);
+    };
+
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    const onChange = (nextStepIndex: number) => {
+        // Before the last step we jump to the wallet
+        if (nextStepIndex === (tourSteps.length - 1)) {
+            navigate("/wallet");
+        } else if (!location.pathname.endsWith("long")) {
```

Comment:
This asks for troubles or move steps around.
What do you think about adding an additional field to the toursteps such as `target`. Then you can do something more generic like: 
```
        let target = tourSteps.at(nextStepIndex).target;
        if (!location.pathname.endsWith(target)) {
            navigate(target);
        }
```

Reply:
I like the proposed idea and am happy to give it a try.

Note: We actually hardcode `long` in several places in the code already, assuming that is the default route 😬

Reply:
Worked out nice! Code is way cleaner!

The behavior did not change.
I wonder if we can even detect the "we are at the start" in a better way, but could not figure out a good way given that we always just get the `next` index `onChange`.